internal/super/generate.go

```diff
@@ -122,64 +136,60 @@ func generateScript(
 	_ flowkit.Services,
 	state *flowkit.State,
 ) (result command.Result, err error) {
-	return generateNew(args, "script", generateFlags.Directory, logger, state)
-}
-
-func addCDCExtension(name string) string {
-	if strings.HasSuffix(name, ".cdc") {
-		return name
+	options := GeneratorOptions{
+		Directory: "cadence",
+		State:     state,
+		Logger:    logger,
 	}
-	return fmt.Sprintf("%s.cdc", name)
+	generator := NewGenerator(options)
+	err = generator.Create(TemplateMap{"script": args[0]})
+	return nil, err
 }
 
-func stripCDCExtension(name string) string {
-	return strings.TrimSuffix(name, filepath.Ext(name))
+// TemplateMap defines a map of template types to their specific names
+type TemplateMap map[string]string
+
+type GeneratorOptions struct {
+	Directory   string
+	State       *flowkit.State
+	Logger      output.Logger
+	DisableLogs bool
 }
 
-// processTemplate reads a template file from the embedded filesystem and processes it with the provided data
-// If you don't need to provide data, pass nil
-func processTemplate(templatePath string, data map[string]interface{}) (string, error) {
-	templateData, err := templatesFS.ReadFile(templatePath)
-	if err != nil {
-		return "", fmt.Errorf("failed to read template file: %w", err)
-	}
+type Generator struct {
+	Options GeneratorOptions
+}
 
-	tmpl, err := template.New("template").Parse(string(templateData))
-	if err != nil {
-		return "", fmt.Errorf("failed to parse template: %w", err)
+func NewGenerator(options GeneratorOptions) *Generator {
+	return &Generator{
+		Options: options,
 	}
+}
 
-	var executedTemplate bytes.Buffer
-	// Execute the template with the provided data or nil if no data is needed
-	if err = tmpl.Execute(&executedTemplate, data); err != nil {
-		return "", fmt.Errorf("failed to execute template: %w", err)
+func (g *Generator) Create(typeNames TemplateMap) error {
```

Comment:
Is a map keyed by type the best data structure for this?  What happens if I want to generate two scripts?

Reply:
You could do it but we could use an array instead. Updated.